pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TransactionEndToEndTest.java

```diff
@@ -1052,13 +1052,16 @@ public void testTxnTimeOutInClient() throws Exception{
             Assert.assertTrue(e.getCause().getCause() instanceof TransactionCoordinatorClientException
                     .InvalidTxnStatusException);
         }
+        Message<String> message = null;
         try {
-            Message<String> message = consumer.receive();
+            message = consumer.receive();
```

Comment:
It feels like adding a timeout to the method avoids getting blocks all the time. But I'm not sure if we have another mechanism to guarantee that.

Reply:
Ci test will run for up to 30s, and an error will be reported if it times out. Of course, we cannot rely on this.